### DISCUSSION TITLE
CapRover and Github not updating file changes

### DISCUSSION THREAD

**Discussion by ashhulme** [caprover/caprover#1692] [General]
Hi all thanks in advance for the advice, I have set up SSH keys along with captain-definition within my project when I force the build it successfully does it however files from git are not been pulled over to the files within the server I know that GitHub and the server are talking to each other due to it running the build

I've tested the webhook and this is correct
My branch is set to "main" within git and caprover
the app is using Persistent Directories 

the only errors i can see when forcing the build is

[Warning] One or more build-args [CAPROVER_GIT_COMMIT_SHA] were not consumed
Successfully built 2b3e5aae1fc5
Successfully tagged img-captain-XXXX:latest
Build has finished successfully!

Thanks

**Comment by githubsaturn** [comment]
What do you mean by this?

> however files from git are not been pulled over to the files within the server

If the build is successful it has pulled your repo

**Comment by ashhulme** [comment]
Hi thanks for your reply, am I right in thinking that everything within GitHub Should be pulled through onto CapRover?

my docker locally runs Laravel with MySQL and PHPMyAdmin
my docker build on CapRover is just php8.1-alpine as I have another app in CapRover to handle the Database

or am I right in thinking I need to change this so that the docker locally is identical to the live build on the server.

before the GitHub integration, I would manually upload the necessary files to the server from the Laravel app

Thanks

**Reply by githubsaturn** [reply]
If you have a persistent directly those will be overriden by the values that are present on your host. 

Other than that, yes, all files are pulled from the server. In simple terms this is what happens:
- The webhook gets called (either from Github or manually by you forcing it)
- CapRover runes `git clone ....` to clone the new repository
- Then it runs `docker build` on the newly download repo
- Finally, it updates the image with the latest image that it has built.

**Reply by ashhulme** [reply]
thanks @githubsaturn  I found my error I was settings one of the Persistent Directories to the root of my app where I needed to set it to a subfolder within the app instead
